### PR TITLE
add ipyleaflet-gl-vector-layer-plugin

### PR DESCRIPTION
**Pull request by srepmub**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by svniemeijer** [comment]
I confirm to be a maintainer

**Comment by srepmub** [comment]
@conda-forge/staged-recipes sorry, but since there is another PR waiting on this (#18901), it would be great if this one could be merged soon (both quite simple recipes I think?)

File: recipes/ipyleaflet-gl-vector-layer-plugin/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cudnn:                                    # [linux64]
+  - undefined                             # [linux64]
+cuda_compiler_version:                    # [linux64]
+  - None                                  # [linux64]
+docker_image:                             # [linux64]
+  - condaforge/linux-anvil-cos7-x86_64    # [linux64]
+cdt_name:                                 # [linux64]
+  - cos7                                  # [linux64]
```
Comment (ocefpaf) [inline recipes/ipyleaflet-gl-vector-layer-plugin/conda_build_config.yaml:8]: You probably do not need this file.

Comment (ocefpaf) [reply recipes/ipyleaflet-gl-vector-layer-plugin/conda_build_config.yaml]: The error is: `jupyter_packaging==0.12.0 is incompatible with jupyter_packaging~=0.7.9`. That is a bit odd, let me check a few things first.

File: recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml [hunk 2]
```diff
@@ -16,18 +16,16 @@ build:
 
 requirements:
   host:
-    - python >=3.9
+    - python >=3.6
```
Comment (ocefpaf) [inline recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml:19]: According to upstream this should work with >=3.6.

File: recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml [hunk 3]
```diff
@@ -16,18 +16,16 @@ build:
...
     - pip
-    - jupyter-packaging
-    - jupyterlab >=3
+    - setuptools >=40.8.0
+    - jupyter-packaging ~=0.7.9
```
Comment (ocefpaf) [inline recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml:22]: Hopefully by specifying the version here we avoid the error we are seeing.

File: recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml [hunk 4]
```diff
@@ -16,18 +16,16 @@ build:
...
+    - jupyterlab >=3.0.0
     - yarn
-
   run:
-    - python >=3.9
-    - traittypes
-    - traitlets
```
Comment (ocefpaf) [inline recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml]: traittypes and traitlets are not listed as dependencies upstream. Did you add them b/c they are optional or something?

File: recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml [hunk 5]
```diff
@@ -16,18 +16,16 @@ build:
...
-    - ipyleaflet =0.15.0
-    - numpy
+    - python >=3.6
+    - ipywidgets >=7.6.0
+    - ipyleaflet >=0.15.0
```
Comment (ocefpaf) [inline recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml:28]: Upstream says` >=`, not `=0.15.0`.

File: recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml (hunk 4 above)
Comment (ocefpaf) [reply recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml]: @srepmub if you are OK with mu changes we can merge this one.

Comment (srepmub) [reply recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml]: thanks! the traitlets/traittypes are imported from ipyleaflet_gl_vector_layer/plugin.py.. which should break the import test right, unless they are pulled in by something else I guess. the rest looks fine to me ^^

Comment (srepmub) [reply recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml]: I added these back in, also numpy, as they are imported from the python code. will see if I can get those added upstream for the next release.

Comment (srepmub) [reply recipes/ipyleaflet-gl-vector-layer-plugin/meta.yaml]: added to upstream main.

**Comment by srepmub** [comment]
https://github.com/orgs/conda-forge/teams/staged-recipes please review changes

**Pull request merged**